File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/governance/GovernanceGetStakingInfoTest.java
package opensdk.sdk.apis.governance;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.GovernanceGetStakingInfoResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Governance RPC Test")
public class GovernanceGetStakingInfoTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC governance_getStakingInfo")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String blockNum = "latest";

        GovernanceGetStakingInfoResponse response = sdk.governance.getStakingInfo(blockNum).send();
        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/governance/GovernanceGetStakingInfo.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.GovernanceApi(new OpenSdk.ApiClient(RPC));

describe('governance_getStakingInfo API', () => {
    test('should return governance_getStakingInfo', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            if (data !== null) {
                expect(typeof data.blockNum === 'number' || /^0x[0-9a-fA-F]+$/.test(data.blockNum)).toBe(true);
            }
            done();
        };

        const blockNumber = "latest";

        sdk.getStakingInfo(blockNumber, {}, callbackOne);
    });
});

